index.js

```diff
@@ -162,11 +162,13 @@ Client.prototype._pollConfig = function () {
     opts.headers['If-None-Match'] = this._conf.lastConfigEtag
   }
 
+  const onError = err => {
+    this._scheduleNextConfigPoll()
+    this.emit('request-error', err)
+  }
+
   const req = this._transport.get(opts, res => {
-    res.on('error', err => {
-      // Not sure this event can ever be emitted, but just in case
-      res.destroy(err)
-    })
+    res.on('error', onError) // Not sure this event can ever be emitted, but just in case
```

Comment:
This reply is for all the comments related to that change.

This is what I was trying to explain initially, but I admit this is hard to explain... So one more try!
The old code calls `res.destroy` which will:
 - Kill the socket if keepAlives are disabled, which will fire a `error` event on the request, which will make the Client emit a `request-error` event. Yes, the `request-error` event was sent, there are tests that proves it but again, when `keepAlives` are disabled.
 - Do nothing at all when `keepAlives` are enabled, which is expected because the socket stays open for future request handling. In this case the `request-error` event logic wasn't triggered at all. That's why I'm manually calling it now
 
keepAlives are enabled by default but tests were passing because the Agent was ignored in the code, as you spotted in a previous review.

So my changes standardize the error handling, no matter if keepAlives are enabled or not, making the tests green.

I hope that my intent and the effects of the code are clearer now :)